From 53c2b4387cacc0a5a065c54a6c14fdefdbd08231 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 22 Sep 2021 12:14:52 +0530
Subject: [PATCH] Bluetooth: btusb: Add reset on close quirk for Intel adapters

Intel Bluetooth adapter is not exiting the loopback mode on bluetooth
socket closure. So, when bluetooth socket is opened again, bluetooth is
not working.

As per Bluetooth core specification, bluetooth adapter should exit loop
back mode on hci reset. Add HCI quirk reset to make sure adapter exits
loopback mode on bluetooth socket closure.

Tracked-On: OAM-96067
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Amrita Raju <amrita.raju@intel.com>
---
 drivers/bluetooth/btusb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/bluetooth/btusb.c b/drivers/bluetooth/btusb.c
index e34599fef79c..b8bce1e3b875 100644
--- a/drivers/bluetooth/btusb.c
+++ b/drivers/bluetooth/btusb.c
@@ -3758,6 +3758,7 @@ static int btusb_probe(struct usb_interface *intf,
 		hdev->send = btusb_send_frame_intel;
 		hdev->cmd_timeout = btusb_intel_cmd_timeout;
 
+		reset = false;
 		if (id->driver_info & BTUSB_INTEL_NO_WBS_SUPPORT)
 			btintel_set_flag(hdev, INTEL_ROM_LEGACY_NO_WBS_SUPPORT);
 
-- 
2.37.0

